loginserver/client.cpp

```diff
@@ -583,6 +588,73 @@ void Client::DoSuccessfulLogin(
 	m_client_status = cs_logged_in;
 }
 
+void Client::SendExpansionPacketData(PlayerLoginReply_Struct& plrs)
+{
+	int default_ruleset = 1;
+	SerializeBuffer buf;
+	//from eqlsstr_us.txt id of each expansion, excluding 'Everquest'
+	int ExpansionLookup[20] = { 3007, 3008, 3009, 3010,	3012,
+								3014, 3031, 3033, 3036, 3040,
+								3045, 3046, 3047, 3514, 3516,
+								3518, 3520, 3522, 3524 };
+
+
+	//Get the active expansion from the database Rule:WorldExpansionSettings.  Do this only once.  Requires restart to effect change
+	if (!owned_expansion) {
+		std::string r_name = RuleManager::Instance()->GetRulesetName(server.db, default_ruleset);
+		if (r_name.size() > 0) {
+			RuleManager::Instance()->LoadRules(server.db, r_name.c_str(), false);
+			expansion = RuleManager::Instance()->GetIntRule(RuleManager::Int__ExpansionSettings);
+			owned_expansion = (expansion << 1) | 1;
+		}
+	}
+
+	if (m_client_version == cv_sod) {
+
+		// header info of packet.  Requires OP_LoginExpansionPacketData=0x0031 to be in login_opcodes_sod.conf
+		buf.WriteInt32(0x00);
+		buf.WriteInt32(0x01);
+		buf.WriteInt16(0x00);
+		buf.WriteInt32(19); //number of expansions to include in packet
+
+		//generate expansion data
+		for (int i = 0; i < 19; i++)
+		{
+			buf.WriteInt32(i);													//sequenctial number
+			buf.WriteInt32((expansion & (1 << i)) == (1 << i) ? 0x01 : 0x00);	//1 own 0 not own
+			buf.WriteInt8(0x00);
+			buf.WriteInt32(ExpansionLookup[i]);									//from eqlsstr_us.txt
+			buf.WriteInt32(0x179E);												//from eqlsstr_us.txt for buttons/order
+			buf.WriteInt32(0xFFFFFFFF);											//end identification
+			buf.WriteInt8(0x0);													//force order window to appear 1 appear 0 not appear
+			buf.WriteInt8(0x0);
+			buf.WriteInt32(0x0000);
+			buf.WriteInt32(0x0000);
+			buf.WriteInt32(0xFFFFFFFF);
+		}
+
+		auto out = std::make_unique<EQApplicationPacket>(OP_LoginExpansionPacketData, buf);
+		m_connection->QueuePacket(out.get());
+
+	}
+	else if (m_client_version == cv_titanium)
+	{
+		if (expansion > 1023)
```

Comment:
Changing this to `>= EQ::expansions::bitPoR` would make this a little clearer

Reply:
updated to use EQ::expansions